### DISCUSSION TITLE
Unable to Authenticate

### DISCUSSION THREAD

**Discussion by CDanaher** [cockpit-project/cockpit#16338] [Q&A] [answered]
I have set up my cockpit with a non-root account.  I am able to do everything I need except when I go to look at updates I have to switch to the root account or it errors with Unable to Authenticate.  Is there a specific elevated group or config that I need to tweak to be able to do this on a non-root account?

**Comment by martinpitt** [comment]
Cockpit uses `sudo` to elevate privileges. Depending on your distribution, this requires your user to be in the `sudo` (Debian/Ubuntu) or `wheel` (Fedora, RHEL, CentOS) group, or of course you could also have custom sudo rules. There is nothing Cockpit specific about that -- if you can run `sudo` in SSH or the desktop, you should be able to switch to administrator mode in cockpit.

**Reply by CDanaher** [reply]
The account is in the correct group and can run sudo commands.
`User ***** may run the following commands on *****:
    (ALL) ALL`

But it still fails at obtaining authentication just on the update option. 


This may be working as intended but it looks like when I change the password in the cockpit console instead of on the server for the service account, then it works.

**Comment by martinpitt** [comment, marked as answer]
This question is quite old, and inconclusive -- I wouldn't know why the Updates page behaves different. It tries to connect to PackageKit as root if it can (i. e. you are in Administrator mode). Please send a screenshot if this still happens with the current version.

**Reply by matt-keys** [reply]
I've ran into this today. To explain adequately I'll need to cover recent history. I was running 4 Debian 10 servers, nfs, vhost, vhost1, vhost2. All are running cockpit, all were working without issue while using Debian 10. 

I had a disk failure on nfs in which I needed to order a new disk and reinstall. I reinstalled with Debian 11. There is a cockpit version difference between 11 and 10:

Deb 11 ..
root@nfs:~# apt-cache show cockpit
Package: cockpit
Version: 239-1

Deb 10 ..
root@vhost1:~# apt-cache show cockpit
Package: cockpit
Version: 188-1

The non-root user is in sudo group on all 4 servers. I can ssh as this user to each host and 'sudo su -' without issue. 

When running 'software updates' from the Debian 11 cockpit to the Debian 10 hosts, /var/log/messages on the Debian 10 hosts indicates ... 

`May 14 08:36:27 vhost1 cockpit-bridge[7077]: polkit-agent-helper-1: pam_authenticate failed: Authentication failure
May 14 08:36:27 vhost1 cockpit-bridge[7077]: Error executing command as another user: Not authorized
May 14 08:36:27 vhost1 cockpit-bridge[7077]: This incident has been reported.
May 14 08:36:29 vhost1 cockpit-bridge[7077]: Sorry, try again.
May 14 08:36:30 vhost1 cockpit-bridge[7077]: Sorry, try again.
May 14 08:36:33 vhost1 cockpit-bridge[7077]: sudo: 3 incorrect password attempts
May 14 08:36:35 vhost1 cockpit-bridge[7077]: polkit-agent-helper-1: pam_authenticate failed: Authentication failure
May 14 08:37:46 vhost1 udisksd[781]: Error loading modules: Error opening directory “/usr/lib/x86_64-linux-gnu/udisks2/modules”: No such file or directory
May 14 08:37:47 vhost1 cockpit-bridge[7077]: No journal files were opened due to insufficient permissions.
`

On the deb10 host (vhost1), I ran 'tail -f /var/log/messages /var/log/syslog /var/log/auth.log' , logged into deb11 (nfs) cockpit, then navigated to the debian10 host within cockpit. The 'system' tab it defaults to automatically checks for updates. The output of the tail ..

`----begin----
May 14 09:08:31 vhost1 sshd[8043]: Accepted publickey for matt from 192.168.1.5 port 42724 ssh2: RSA SHA256:CGNT3c6wK3hxelpWOFnyYHe9NaIvK4ejHtUYarZDeeA
May 14 09:08:31 vhost1 sshd[8043]: pam_unix(sshd:session): session opened for user matt by (uid=0)
May 14 09:08:31 vhost1 systemd-logind[779]: New session 13 of user matt.

==> syslog <==
May 14 09:08:31 vhost1 systemd[1]: Started Session 13 of user matt.

==> auth.log <==
May 14 09:08:31 vhost1 polkitd(authority=local): Registered Authentication Agent for unix-session:13 (system bus name :1.488 [cockpit-bridge], object path /org/freedesktop/PolicyKit1/AuthenticationAgent, locale en_US.UTF-8)

==> syslog <==
May 14 09:08:31 vhost1 dbus-daemon[808]: [system] Activating via systemd: service name='org.freedesktop.hostname1' unit='dbus-org.freedesktop.hostname1.service' requested by ':1.506' (uid=1000 pid=8050 comm="cockpit-bridge ")
May 14 09:08:31 vhost1 systemd[1]: Starting Hostname Service...
May 14 09:08:31 vhost1 dbus-daemon[808]: [system] Activating via systemd: service name='org.freedesktop.timedate1' unit='dbus-org.freedesktop.timedate1.service' requested by ':1.507' (uid=1000 pid=8050 comm="cockpit-bridge ")
May 14 09:08:31 vhost1 systemd[1]: Starting Time & Date Service...
May 14 09:08:31 vhost1 dbus-daemon[808]: [system] Activating via systemd: service name='org.freedesktop.realmd' unit='realmd.service' requested by ':1.507' (uid=1000 pid=8050 comm="cockpit-bridge ")
May 14 09:08:31 vhost1 systemd[1]: Starting Realm and Domain Configuration...

==> auth.log <==
May 14 09:08:31 vhost1 realmd[8068]: Loaded settings from: /usr/lib/realmd/realmd-defaults.conf /usr/lib/realmd/realmd-distro.conf
May 14 09:08:31 vhost1 realmd[8068]: holding daemon: startup
May 14 09:08:31 vhost1 realmd[8068]: starting service
May 14 09:08:31 vhost1 realmd[8068]: connected to bus
May 14 09:08:31 vhost1 realmd[8068]: released daemon: startup

==> syslog <==
May 14 09:08:31 vhost1 dbus-daemon[808]: [system] Successfully activated service 'org.freedesktop.realmd'

==> auth.log <==
May 14 09:08:31 vhost1 realmd[8068]: claimed name on bus: org.freedesktop.realmd
May 14 09:08:31 vhost1 polkit-agent-helper-1[8067]: pam_unix(polkit-1:auth): authentication failure; logname= uid=1000 euid=0 tty= ruser=matt rhost=  user=matt
May 14 09:08:31 vhost1 realmd[8068]: client using service: :1.507

==> syslog <==
May 14 09:08:31 vhost1 systemd[1]: Started Realm and Domain Configuration.

==> auth.log <==
May 14 09:08:31 vhost1 realmd[8068]: holding daemon: :1.507

==> syslog <==
May 14 09:08:31 vhost1 dbus-daemon[808]: [system] Successfully activated service 'org.freedesktop.timedate1'
May 14 09:08:31 vhost1 systemd[1]: Started Time & Date Service.
May 14 09:08:31 vhost1 dbus-daemon[808]: [system] Successfully activated service 'org.freedesktop.hostname1'
May 14 09:08:31 vhost1 systemd[1]: Started Hostname Service.
May 14 09:08:34 vhost1 cockpit-bridge[8050]: polkit-agent-helper-1: pam_authenticate failed: Authentication failure

==> messages <==
May 14 09:08:34 vhost1 cockpit-bridge[8050]: polkit-agent-helper-1: pam_authenticate failed: Authentication failure

==> auth.log <==
May 14 09:08:34 vhost1 polkitd(authority=local): Operator of unix-session:13 FAILED to authenticate to gain authorization for action org.cockpit-project.cockpit.root-bridge for unix-process:8050:206821 [cockpit-bridge] (owned by unix-user:matt)
May 14 09:08:34 vhost1 pkexec[8058]: matt: Error executing command as another user: Not authorized [USER=root] [TTY=unknown] [CWD=/run/user/1000] [COMMAND=/usr/bin/cockpit-bridge --privileged]

==> syslog <==
May 14 09:08:34 vhost1 cockpit-bridge[8050]: Error executing command as another user: Not authorized

==> messages <==
May 14 09:08:34 vhost1 cockpit-bridge[8050]: Error executing command as another user: Not authorized
May 14 09:08:34 vhost1 cockpit-bridge[8050]: This incident has been reported.

==> syslog <==
May 14 09:08:34 vhost1 cockpit-bridge[8050]: This incident has been reported.

==> auth.log <==
May 14 09:08:34 vhost1 sudo: pam_unix(sudo:auth): authentication failure; logname= uid=1000 euid=0 tty= ruser=matt rhost=  user=matt

==> syslog <==
May 14 09:08:36 vhost1 cockpit-bridge[8050]: Sorry, try again.

==> messages <==
May 14 09:08:36 vhost1 cockpit-bridge[8050]: Sorry, try again.

==> syslog <==
May 14 09:08:37 vhost1 cockpit-bridge[8050]: Sorry, try again.

==> messages <==
May 14 09:08:37 vhost1 cockpit-bridge[8050]: Sorry, try again.

==> auth.log <==
May 14 09:08:40 vhost1 sudo:     matt : 3 incorrect password attempts ; TTY=unknown ; PWD=/run/user/1000 ; USER=root ; COMMAND=/usr/bin/cockpit-bridge --privileged

==> syslog <==
May 14 09:08:40 vhost1 cockpit-bridge[8050]: sudo: 3 incorrect password attempts

==> messages <==
May 14 09:08:40 vhost1 cockpit-bridge[8050]: sudo: 3 incorrect password attempts

==> syslog <==
May 14 09:08:40 vhost1 dbus-daemon[808]: [system] Activating via systemd: service name='org.freedesktop.PackageKit' unit='packagekit.service' requested by ':1.507' (uid=1000 pid=8050 comm="cockpit-bridge ")
May 14 09:08:40 vhost1 systemd[1]: Starting PackageKit Daemon...
May 14 09:08:40 vhost1 PackageKit: daemon start
May 14 09:08:40 vhost1 dbus-daemon[808]: [system] Successfully activated service 'org.freedesktop.PackageKit'
May 14 09:08:40 vhost1 systemd[1]: Started PackageKit Daemon.
May 14 09:08:40 vhost1 PackageKit: get-updates transaction /1299_cbddbede from uid 1000 finished with success after 277ms
`

Cockpit displayed ..

Operating System
Debian GNU/Linux 10 (buster)
System Up To Date

I then clicked 'software updates' while monitoring the tail ...

`==> syslog <==
May 14 09:13:31 vhost1 PackageKit: get-updates transaction /1305_bcdacede from uid 1000 finished with success after 267ms
May 14 09:13:32 vhost1 PackageKit: get-updates transaction /1306_aabeebac from uid 1000 finished with success after 286ms

==> auth.log <==
May 14 09:13:32 vhost1 PackageKit: uid 1000 is trying to obtain org.freedesktop.packagekit.system-sources-refresh auth (only_trusted:0)
May 14 09:13:32 vhost1 polkit-agent-helper-1[8160]: pam_unix(polkit-1:auth): authentication failure; logname= uid=1000 euid=0 tty= ruser=matt rhost=  user=matt

==> syslog <==
May 14 09:13:33 vhost1 cockpit-bridge[8050]: polkit-agent-helper-1: pam_authenticate failed: Authentication failure

==> messages <==
May 14 09:13:33 vhost1 cockpit-bridge[8050]: polkit-agent-helper-1: pam_authenticate failed: Authentication failure

==> auth.log <==
May 14 09:13:33 vhost1 polkitd(authority=local): Operator of unix-session:13 FAILED to authenticate to gain authorization for action org.freedesktop.packagekit.system-sources-refresh for system-bus-name::1.517 [cockpit-bridge] (owned by unix-user:matt)
May 14 09:13:33 vhost1 PackageKit: uid 1000 failed to obtain auth
`

Cockpit displayed .. 

Loading available updates failed
Failed to obtain authentication.

Regards,
Matt

**Reply by matt-keys** [reply]
non root user can ssh to all hosts and sudo su - without any issue:

matt@nfs:~$ ssh 192.168.1.2
Linux vhost1 4.19.0-24-amd64 #1 SMP Debian 4.19.282-1 (2023-04-29) x86_64

The programs included with the Debian GNU/Linux system are free software;
the exact distribution terms for each program are described in the
individual files in /usr/share/doc/*/copyright.

Debian GNU/Linux comes with ABSOLUTELY NO WARRANTY, to the extent
permitted by applicable law.
Web console: https://vhost1.home.mattkeys.net:9090/ or https://192.168.1.2:9090/

You have new mail.
Last login: Mon May 15 09:42:05 2023 from 192.168.1.5
matt@vhost1:~$ sudo su -
[sudo] password for matt:
root@vhost1:~# exit
logout
matt@vhost1:~$ exit
logout
Connection to 192.168.1.2 closed.
matt@nfs:~$ ssh 192.168.1.3
Linux vhost2 4.19.0-23-amd64 #1 SMP Debian 4.19.269-1 (2022-12-20) x86_64

The programs included with the Debian GNU/Linux system are free software;
the exact distribution terms for each program are described in the
individual files in /usr/share/doc/*/copyright.

Debian GNU/Linux comes with ABSOLUTELY NO WARRANTY, to the extent
permitted by applicable law.
Web console: https://vhost2.home.mattkeys.net:9090/ or https://192.168.1.3:9090/

You have new mail.
Last login: Mon May 15 09:42:21 2023 from 192.168.1.5
matt@vhost2:~$ sudo su -
[sudo] password for matt:
root@vhost2:~# exit
logout
matt@vhost2:~$ exit
logout
Connection to 192.168.1.3 closed.
matt@nfs:~$ ssh 192.168.1.4
Linux vhost 4.19.0-23-amd64 #1 SMP Debian 4.19.269-1 (2022-12-20) x86_64

The programs included with the Debian GNU/Linux system are free software;
the exact distribution terms for each program are described in the
individual files in /usr/share/doc/*/copyright.

Debian GNU/Linux comes with ABSOLUTELY NO WARRANTY, to the extent
permitted by applicable law.
Web console: https://vhost.home.mattkeys.net:9090/ or https://192.168.1.4:9090/

You have new mail.
Last login: Mon May 15 09:42:32 2023 from 192.168.1.5
matt@vhost:~$ sudo su -
[sudo] password for matt:
root@vhost:~# exit
logout
matt@vhost:~$ exit
logout
Connection to 192.168.1.4 closed.
matt@nfs:~$
